### PR TITLE
add sdn migration ovn job

### PR DESCRIPTION
**Pull request by zhaozhanqi**

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml [hunk 1]
```diff
@@ -275,6 +275,17 @@ tests:
     test:
     - ref: openshift-extended-test
     workflow: cucushift-installer-rehearse-aws-ipi-private
+- as: sdn-migration-ovn-aws-ipi
```
Comment (kuiwang02) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml:278]: @zhaozhanqi the name better be `e2e-aws-ipi-sdn-migration-ovn`

Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml:278]: ok, will fix this

Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml]: done

**Comment by anuragthehatter** [comment]
We possibly can add @weliang1 as OWNER as he will be leading this feature mostly. Make sure to run ```make registry-metadata``` post that

**Comment by weliang1** [comment]
@zhaozhanqi Some dumb questions:
1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?
2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct?
3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?

**Comment by zhaozhanqi** [comment]
> @zhaozhanqi Some dumb questions: 1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?

openshift-openshift-tests-private-release-4.11__nightly.yaml is in `ci-operator/config/openshift/openshift-tests-private/` and we can config kind of testing in this file(like CI test profile in jenkins), 
this file `openshift-openshift-tests-private-release-4.11-periodics.yaml` is generated by auto when you adding one config( eg. this PR sdn-migartion)  and then do `make jobs`.  we do not need to update this file by manual. 

> 2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct? 

yes, this one for aws-ipi

> 3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?

this PR or jobs will not do upgrade testing,  it's only invoked upgrade pre --> migration --> post test data.

**Comment by weliang1** [comment]
> > @zhaozhanqi Some dumb questions: 1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?
> 
> openshift-openshift-tests-private-release-4.11__nightly.yaml is in `ci-operator/config/openshift/openshift-tests-private/` and we can config kind of testing in this file(like CI test profile in jenkins), this file `openshift-openshift-tests-private-release-4.11-periodics.yaml` is generated by auto when you adding one config( eg. this PR sdn-migartion) and then do `make jobs`. we do not need to update this file by manual.
> 
> > 2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct?
> 
> yes, this one for aws-ipi
> 
> > 3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?
> 
> this PR or jobs will not do upgrade testing, it's only invoked upgrade pre --> migration --> post test data.

@zhaozhanqi Make sense, thanks for the explanations. 
One more question, do we still need a similar PR (may be named like add sdn to ovn workflow and cluster profile for aws cluster) as https://github.com/openshift/release/pull/28087/files for enabling sdn-ovn-migration in @aws-ipi cluster?

**Comment by zhaozhanqi** [comment]
> > > @zhaozhanqi Some dumb questions: 1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?
> > 
> > 
> > openshift-openshift-tests-private-release-4.11__nightly.yaml is in `ci-operator/config/openshift/openshift-tests-private/` and we can config kind of testing in this file(like CI test profile in jenkins), this file `openshift-openshift-tests-private-release-4.11-periodics.yaml` is generated by auto when you adding one config( eg. this PR sdn-migartion) and then do `make jobs`. we do not need to update this file by manual.
> > > 2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct?
> > 
> > 
> > yes, this one for aws-ipi
> > > 3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?
> > 
> > 
> > this PR or jobs will not do upgrade testing, it's only invoked upgrade pre --> migration --> post test data.
> 
> @zhaozhanqi Make sense, thanks for the explanations. One more question, do we still need a similar PR (may be named like add sdn to ovn workflow and cluster profile for aws cluster) as https://github.com/openshift/release/pull/28087/files for enabling sdn-ovn-migration in @aws-ipi cluster?

yes,  we still need one aws-ipi because we need to check all ovn cases with tag @openshift-ovnkubenetes after migration to ovn. 

this is my thought: 
1. we just need only this one profile to make sure the prepared test data can be worked well after migration
2. we can create workflows like https://github.com/openshift/release/pull/28087 for each supported platform including IPI/UPI to make sure they are working well 
WDYT?

**Comment by zhaozhanqi** [comment]
> > > > > @zhaozhanqi Some dumb questions: 1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?
> > > > 
> > > > 
> > > > openshift-openshift-tests-private-release-4.11__nightly.yaml is in `ci-operator/config/openshift/openshift-tests-private/` and we can config kind of testing in this file(like CI test profile in jenkins), this file `openshift-openshift-tests-private-release-4.11-periodics.yaml` is generated by auto when you adding one config( eg. this PR sdn-migartion) and then do `make jobs`. we do not need to update this file by manual.
> > > > > 2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct?
> > > > 
> > > > 
> > > > yes, this one for aws-ipi
> > > > > 3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?
> > > > 
> > > > 
> > > > this PR or jobs will not do upgrade testing, it's only invoked upgrade pre --> migration --> post test data.
> > > 
> > > 
> > > @zhaozhanqi Make sense, thanks for the explanations. One more question, do we still need a similar PR (may be named like add sdn to ovn workflow and cluster profile for aws cluster) as https://github.com/openshift/release/pull/28087/files for enabling sdn-ovn-migration in @aws-ipi cluster?
> > 
> > 
> > yes, we still need one aws-ipi because we need to check all ovn cases with tag @openshift-ovnkubenetes after migration to ovn.
> > this is my thought:
> > 
> > 1. we just need only this one profile to make sure the prepared test data can be worked well after migration
> > 2. we can create workflows like [[WIP]add sdn to ovn workflow and cluster profile for vsphere #28087](https://github.com/openshift/release/pull/28087) for each supported platform including IPI/UPI to make sure they are working well
> >    WDYT?
> 
> That's sounds good! Can we try to make this plan works in one cluster first? either vsphere-ipi(PR28087) or aws-ipi(PR28562)

yes, PR28562 is for 1. 
You can try to create another workflow like PR28087 on IPI aws for 2.

**Comment by weliang1** [comment]
> > > > > > @zhaozhanqi Some dumb questions: 1.What is the difference of purpose to run openshift-openshift-tests-private-release-4.11__nightly.yaml and openshift-openshift-tests-private-release-4.11-periodics.yaml?
> > > > > 
> > > > > 
> > > > > openshift-openshift-tests-private-release-4.11__nightly.yaml is in `ci-operator/config/openshift/openshift-tests-private/` and we can config kind of testing in this file(like CI test profile in jenkins), this file `openshift-openshift-tests-private-release-4.11-periodics.yaml` is generated by auto when you adding one config( eg. this PR sdn-migartion) and then do `make jobs`. we do not need to update this file by manual.

5 files are changed in PR28562, 12 files are changed in PR28087, could you let me know which files are generated by auto and not manually?

> > > > > > 2.This PR only enable sdn-ovn-migration in @aws-ipi cluster, correct?
> > > > > 
> > > > > 
> > > > > yes, this one for aws-ipi
> > > > > > 3.This PR will run sdn-ovn-migration testing desperately from upgrading testing, not as Jenkins flow does, correct?
> > > > > 
> > > > > 
> > > > > this PR or jobs will not do upgrade testing, it's only invoked upgrade pre --> migration --> post test data.
> > > > 
> > > > 
> > > > @zhaozhanqi Make sense, thanks for the explanations. One more question, do we still need a similar PR (may be named like add sdn to ovn workflow and cluster profile for aws cluster) as https://github.com/openshift/release/pull/28087/files for enabling sdn-ovn-migration in @aws-ipi cluster?
> > > 
> > > 
> > > yes, we still need one aws-ipi because we need to check all ovn cases with tag @openshift-ovnkubenetes after migration to ovn.
> > > this is my thought:
> > > 
> > > 1. we just need only this one profile to make sure the prepared test data can be worked well after migration
> > > 2. we can create workflows like [[WIP]add sdn to ovn workflow and cluster profile for vsphere #28087](https://github.com/openshift/release/pull/28087) for each supported platform including IPI/UPI to make sure they are working well
> > >    WDYT?
> > 
> > 
> > That's sounds good! Can we try to make this plan works in one cluster first? either vsphere-ipi(PR28087) or aws-ipi(PR28562)
> 
> yes, PR28562 is for 1. You can try to create another workflow like PR28087 on IPI aws for 2.

Using PR28087 as a good example for rest of workflows, better to wait for all the comments get updated and commits merged for PR28082

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml [hunk 2]
```diff
@@ -330,6 +330,17 @@ tests:
     test:
     - ref: openshift-extended-test
     workflow: cucushift-installer-rehearse-aws-ipi-private
+- as: e2e-aws-ipi-sdn-migration-ovn
```
[the remark this answers is not in the document]
Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml:333]: @liangxia sorry, I did not get you. what do you mean `sort the jobs`?

[the remark this answers is not in the document]
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml:333]: Then names (the part which is after `-as:`) should be in alphanumeric order.

[the remark this answers is not in the document]
Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly.yaml]: ah, ok,  got it.  updated this.  Please take a look. thanks

**Comment by kuiwang02** [comment]
@zhaozhanqi after you modify the config file every time, please execute `make update` in your local and commit the generated change. it should avoid the `ci-operator-config-metadata` fail because it does many checking and updated including `make ci-operator-config`

**Comment by zhaozhanqi** [comment]
> @zhaozhanqi after you modify the config file every time, please execute `make update` in your local and commit the generated change. it should avoid the `ci-operator-config-metadata` fail because it does many checking and updated including `make ci-operator-config`

ok, thanks, just squash the commits. maybe miss this

**Pull request merged**